### ISSUE TITLE
`Assertions.assertNotNull()` should return the non-null object

### ISSUE THREAD

**Issue by garydgregory** [junit-team/junit-framework#4015]
As a developer, I like to write some test code like:
```java
import static org.junit.jupiter.api.Assertions.assertNotNull;
...
Foo foo = assertNotNull(bar.doSomething()).getFoo();
```
## Deliverables

- An `Assertions.assertNotNull()` that returns its input and not `void`.
- You could make the same argument for `assertSame()`

**Comment by scordio** [comment]
If using AssertJ is an option, with version 3.27.0 it will be possible to write the following:

```java
import static org.assertj.core.api.Assertions.assertThat;
...
Foo foo = assertThat(bar.doSomething()).isNotNull().actual().getFoo();
```

See assertj/assertj#3489.

**Comment by garydgregory** [comment]
Hello @scordio 

Thank you for your reply.

I'll be honest with you: I'll never write this kind of extra verbose code. I feel that this obfuscates what the test is trying to show. 

The reason I gave this example is because I want _less_ code, not more ;-) I'll stick with the straightforward for now :)

**Comment by scordio** [comment]
Totally understandable 😉 a pure non-null assertion is not the best showcase for that, but users find it handy for more complex use cases.

As with everything, YMMV 🙂

**Comment by garydgregory** [comment]
And do I have miles! 😜

**Comment by marcphilipp** [comment]
Changing the result type of `assertNull` would break backward (binary) compatibility. You could use `java.util.Objects.requireNonNull` instead.

**Comment by garydgregory** [comment]
I agree BC is key, this would be for 6.0. Nice idea using requireNonNull 👍

**Comment by bjmi** [comment]
The _assertNotNull_ is verbose and superfluous. Just write
`Foo foo = bar.doSomething().getFoo();`
and [Helpful NullPointerExceptions](https://openjdk.org/jeps/358) show the null cause.

**Comment by garydgregory** [comment]
> The _assertNotNull_ is verbose and superfluous. Just write `Foo foo = bar.doSomething().getFoo();` and [Helpful NullPointerExceptions](https://openjdk.org/jeps/358) show the null cause.

Hello @bjmi 
[JEP 358](https://openjdk.org/jeps/358) is irrelevant here since it requires Java 14 or above. It _is_ a great improvement though! 👍

**Comment by marcphilipp** [comment]
**Team decision:** Since JUnit 6 requires Java 17, we recommend just writing `bar.doSomething().getFoo()` from now on or using `requireNonNull` to avoid any nullability-related IDE/compiler warnings/errors.
